containers/cpp-mariadb/test-project/main.cpp

```diff
@@ -0,0 +1,68 @@
+/*-------------------------------------------------------------------------------------------------------------
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See https://go.microsoft.com/fwlink/?linkid=2090316 for license information.
+ *-------------------------------------------------------------------------------------------------------------*/
+
+#include <iostream>
+#include <mariadb/conncpp.hpp>
+
+using namespace std;
+
+int main()
+{
+    cout << "Hello, Remote World!" << endl;
+
+    try
+    {
+        string databasename = getenv("MARIADB_DATABASE");
+        string password = getenv("MARIADB_PASSWORD");
+        string username = getenv("MARIADB_USER");
+        string hostname = getenv("MARIADB_DATABASE");
+        string jointURL = "jdbc:mariadb://" + hostname + "/" + databasename;
+
+        // Instantiate Driver
+        sql::Driver* driver = sql::mariadb::get_driver_instance();
+
+        cout << "DB Connecting" << endl;
+
+        // Configure Connection
+        sql::SQLString url(jointURL);
+        sql::Properties properties({{"user", username}, {"password", password}});
+
+        // Establish Connection
+        unique_ptr<sql::Connection> conn(driver->connect(url, properties));
+
+        cout << "DB Executing" << endl;
+
+        // Create a new Statement
+        sql::Statement *stmt;
+        // Create resultSet
+        sql::ResultSet *res;
+
+        string query = "show databases "
+                        "where `database` not in " 
+                        "('information_schema', 'performance_schema');";
+
+        stmt = conn->createStatement();
+        res = stmt->executeQuery(query);
+
+        cout << "Cluster has the following user created databases" << endl;
+
+        while(res -> next())
+        {
+            cout << res->getString(1) << endl;
+        }
+
+        cout << "DB Success" << endl;
+
+        delete res;
+        delete stmt;
+        conn.release();
```

Comment:
Did you mean to call `conn.close()` here? 

`release()` on releases the unique_ptr's ownership of the connection. If I read the docs correctly, you'd have to call `delete` on the returned connection pointer to free up the memory of the underlying object. 

https://en.cppreference.com/w/cpp/memory/unique_ptr/release